cmd/debugmailbox/src/lib.rs

```diff
@@ -344,6 +358,58 @@ fn debugmailboxcmd(context: &mut ExecutionContext) -> Result<()> {
 
             let _ = write_req(&mut iface, &dm_port, DMCommand::ExitDM, &[])?;
         }
+        DebugMailboxCmd::AuthOnline { key_name, beacon } => {
+            // Get the challenge from the chip.
+            alive(&mut iface, &dm_port, true)?;
+            let dac = write_req(
+                &mut iface,
+                &dm_port,
+                DMCommand::DebugChallenge,
+                &[],
+            )?;
+
+            // Ask permission-slip to sign it.
+            let mut permslip = Process::new("permslip")
+                .arg("sign")
+                .arg(key_name)
+                .arg("--kind=debug-authn-challenge")
+                .arg(format!("--beacon={beacon}"))
+                .stdin(Stdio::piped())
+                .stdout(Stdio::piped())
+                .spawn()
+                .context("Unable to execute `permslip`, is it in your PATH and executable?")?;
```

Comment:
I almost wish we could just use the permslip client library but given everything is async I think this is probably the best solution. It would be good to document this dependency somewhere, maybe in the doc at the top.

Reply:
So, I started going down that path, but immediately ran into build problems. The main problem is that `permission-slip-client` currently makes no distinction between the client library (which we could use here) and the client binary (which has dependencies that conflict with humility's). This was the fast and easy way to get it working; when we do the split (fix oxidecomputer/permission-slip#103), then we can link in just the client library (and an async runner, etc.).